nix/stack-shell.nix

```diff
@@ -0,0 +1,13 @@
+{ ghc }:
```

Comment:
Hah, didn't know about this one. It's impure (ghc comes from `<nixpkgs>`) and the error message if ghc mismatches is terrible, but it seems better than hardcoding.

Reply:
Yeah it can't be pure, but this is apparently how stack wants it, and it makes sense. It needs to manage the ghcs. That's how it's designed.

If you use two ghcs at the same time, same version, built differently because you're in two projects, you'll get errors that it can't find some libraries.